app/controllers/RiderRankingConnectionController.java

```diff
@@ -0,0 +1,116 @@
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    public CompletionStage<Result> getRiderRankingsByType(long riderStageConnectionId, String rankingType) {
+        return riderRankingRepository.getAllRiderRankingsByType(riderStageConnectionId, rankingType).thenApplyAsync(riderRankings -> {
+            return ok(toJson(riderRankings));
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "IndexOutOfBoundsException":
+                    res = badRequest("No rider rankings are set in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    public CompletionStage<Result> getRiderRankingByRiderAndType(long riderId, String rankingType) {
+        return riderRankingRepository.getRiderRankingByRiderAndType(riderId, rankingType).thenApplyAsync(riderRanking -> {
+            return ok(toJson(riderRanking));
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "IndexOutOfBoundsException":
+                    res = badRequest("No rider ranking is set in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    @BodyParser.Of(BodyParser.Json.class)
+    public CompletionStage<Result> updateRiderRanking(long riderRankingId) {
+        JsonNode json = request().body().asJson();
+        return parseRiderRanking(json, riderRankingId).thenApply(riderRankingRepository::updateRiderRanking).thenApplyAsync(rSC -> {
+            return ok("success");
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "IndexOutOfBoundsException":
+                    res = badRequest("No stage are set in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    private CompletableFuture<RiderRanking> parseRiderRanking(JsonNode json, Long riderRankingId){
+        CompletableFuture<RiderRanking> completableFuture
+                = new CompletableFuture<>();
+
+        Executors.newCachedThreadPool().submit(() -> {
+            try{
+                RiderRanking riderRanking = new RiderRanking();
+                riderRanking.setId(riderRankingId);
+                riderRanking.setRank(json.findPath("rank").asInt());
+                riderRanking.setRankingType(RankingType.valueOf(json.findPath("rankingTyp").textValue()));;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)